docs/lang/articles/advanced/meta.md

```diff
@@ -166,6 +148,18 @@ def func():
   print(3)
 ```
 
+## When to use `ti.static` with for loops
+
+The main reason to use `ti.static` with for loops is to unroll the loop to improve runtime performance (see [Compile-time evaluations](#compile-time-evaluations)).
+
+An unrolling verison is often faster because it reduces branch overhead:
+
+1. It saves the instructions for incrementing the loop counter and checking the end-of-loop condition.
+2. It provides more instructions for the compiler to schedule across the whole iterations.
+
+This performance improvement does not come for free: it also costs longer complie time and generates larger binaries. Especially when the number of iterations is large or the loop body is complex.
```

Comment:
```suggestion
The performance improvement does not come for free but leads to longer compile time and larger binaries, especially when the number of iterations is large or when the loop body is complex.
```

Reply:
I think the suggested change "... but leads to ..." is not as good as the original one?

Reply:
I was thinking about reducing the number of verbs... We can revert to the original form with the colon but I would suggest phrasing it in this way: It takes longer compile time and generates larger binaries, especially when...